### PR TITLE
[Core] Checkout flow and errors improvements

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT
| Doc PR          |

- previous implementation led to confusion; if transition is invalid, then state machine is wrongly configured or wrongly used and it should be shown
- added checking order state before applying some transitions

### CODE REVIEW

File: src/Sylius/Component/Core/Exception/InvalidTransitionException.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class InvalidTransitionException extends SMException
+{
+    /**
+     * @param string $transition
+     */
+    public function __construct($transition)
```
Comment (pamil) [inline src/Sylius/Component/Core/Exception/InvalidTransitionException.php]: DX: Can we also pass an optional list of available states? Or whole `StateMachine` and build even more helpful message? "Transition foo is invalid for bar state machine. Available states are: rynkowski, krawczyk, hołdys"?

Comment (Zales0123) [inline src/Sylius/Component/Core/Exception/InvalidTransitionException.php]: Not a bad idea :+1:

Comment (okwinza) [inline src/Sylius/Component/Core/Exception/InvalidTransitionException.php]: :+1: on this

**Comment by pjedrzejewski** [comment]
Thanks Mateusz!

**Pull request merged**